File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        
if(nums.size()==1){
    if(nums[0]==0){
        return 0;
    }
}
vector<int> system;

int count = 0 ;
int maxCount = 0; 
        for( int i = 0 ; i < nums.size();i++){
                if( nums[i]==1){
                    count++;
                    maxCount = max(maxCount , count);
                    if(i==nums.size()-1){
                          system.push_back(count);
                            count=0;
                            break;
                    }
                }
                else {
                            system.push_back(count);
                            count=0;
                }
        }

int maxSum = maxCount-1;
        for(int i = 0 ; i < system.size()-1;i++){
               maxSum = max(system[i] + system[i+1] , maxSum);
        }

        return maxSum;
    }
};

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubarray = function(nums) {

    let system = [];

    if(nums.length ==1 && nums[0]==0 ){
            return 0;
    }
    let count =0 , maxCount = 0;

        for(let i = 0 ; i <nums.length ; i++){
            if(nums[i]==1){
                count++;
                maxCount = Math.max(maxCount , count);
            }else{
                   system.push(count);
                    count=0;
            }
        }

       if(count!=0){ system.push(count);}

    
    if (system.length === 1 && nums.length === system[0]) {
        return nums.length - 1;
    }

var maxSum = 0;

        for( let i = 0 ; i <system.length-1 ; i++){
            maxSum = Math.max( maxSum , system[i]+system[i+1]);
        }

        return maxSum;


};